template/expand.go

```diff
@@ -38,6 +38,28 @@ func ExpandContainerSpec(t *api.Task) (*api.ContainerSpec, error) {
 	return container, errors.Wrap(err, "expanding hostname failed")
 }
 
+// ExpandNetworks expands templated fields in the network attachment.
+func ExpandNetworks(t *api.Task) ([]*api.NetworkAttachment, error) {
+	networks := make([]*api.NetworkAttachment, len(t.Networks))
+
+	ctx := NewContextFromTask(t)
+	var err error
+
+	for indexNetworks, network := range t.Networks {
+		networks[indexNetworks] = network.Copy()
+		aliases := network.Aliases
+		for indexAliases, alias := range aliases {
+			alias, err = ctx.Expand(alias)
+			if err != nil {
+				return networks, errors.Wrap(err, "expanding alias failed")
+			}
+			aliases[indexAliases] = alias
```

Comment:
This appears to be overwriting the `Aliasses` entry in `t.Networks`, not in the copy that was made above.